hydroflow_datalog/src/grammar.rs

```diff
@@ -54,4 +66,26 @@ pub mod datalog {
         #[rust_sitter::leaf(pattern = r"\s")]
         _whitespace: (),
     }
+
+    #[derive(Debug, Clone)]
+    pub enum BoolOp {
+        Lt(#[rust_sitter::leaf(text = "<")] ()),
+        LtEq(#[rust_sitter::leaf(text = "<=")] ()),
+        Gt(#[rust_sitter::leaf(text = ">")] ()),
+        GtEq(#[rust_sitter::leaf(text = ">=")] ()),
+        Eq(#[rust_sitter::leaf(text = "==")] ()),
+    }
+
+    #[derive(Debug, Clone)]
+    pub struct PredicateExpr {
```

Comment:
Part of me wonders if we should also support arbitrary Rust expressions, maybe using backticks as delimiters. So  you could write expressions like:

```
out(x, y) :- in(x, y), (`my_rust_function(x)` > y)
```

Thoughts? We could leave this to a follow-up PR.

Reply:
Follow up PR. Like to merge something sooner than later at least... this PR has taken like four months so far now :).